pkg/liqoctl/info/localstatus/health.go

```diff
@@ -0,0 +1,131 @@
+}
+
+// HealthChecker collects the info about the local instance of Liqo.
+type HealthChecker struct {
+	info.CheckerBase
+	data Health
+}
+
+// Collect data about the health of the local instance of Liqo.
+func (l *HealthChecker) Collect(ctx context.Context, options info.Options) {
+	var liqoPodsList corev1.PodList
+	if err := options.CRClient.List(ctx, &liqoPodsList, client.InNamespace(options.LiqoNamespace)); err != nil {
+		l.AddCollectionError(fmt.Errorf("unable to get Liqo pods: %w", err))
+		return
+	}
+
+	// Check the status of each pods
+	l.data.Healthy = true
+	l.data.UnhealthyPods = map[string]PodHealthInfo{}
+	liqoPods := liqoPodsList.Items
+	for i := range liqoPods {
+		pod := &liqoPods[i]
+		// Check if one of the pods is not ready, in that case declare Liqo installation as Unhealthy
+		if !l.isPodReady(pod.Status.Conditions) {
+			l.data.Healthy = false
+			healthInfo := PodHealthInfo{
+				Status:          pod.Status.Phase,
+				TotalContainers: len(pod.Status.ContainerStatuses),
+			}
+
+			// Populate the info about restarts and running containers
+			for i := range pod.Status.ContainerStatuses {
+				containerStatus := &pod.Status.ContainerStatuses[i]
+				healthInfo.Restarts += containerStatus.RestartCount
+				if containerStatus.Ready {
+					healthInfo.ReadyContainers++
+				}
+			}
+			l.data.UnhealthyPods[pod.Name] = healthInfo
+		}
+	}
+}
+
+// Format returns the collected data using a user friendly output.
+func (l *HealthChecker) Format(options info.Options) string {
+	main := output.NewRootSection()
+	if l.data.Healthy {
+		main.AddSectionSuccess(fmt.Sprintf("%s    Liqo is healthy", output.CheckMark))
+	} else {
+		main.AddSectionFailure(fmt.Sprintf("%s    Liqo is unhealthy", output.Cross))
+	}
+
+	if len(l.data.UnhealthyPods) > 0 {
+		podsSection := main.AddSection("Unhealthy pods")
+		for podName, podInfo := range l.data.UnhealthyPods {
+			podsSection.AddEntryWithoutStyle(
+				podName,
+				fmt.Sprintf("Status: %v, Ready: %v/%v, Restarts: %v", podInfo.Status, podInfo.ReadyContainers, podInfo.TotalContainers, podInfo.Restarts),
+			)
+		}
+	}
+
+	return main.SprintForBox(options.Printer)
+}
+
+// GetData returns the data collected by the checker.
+func (l *HealthChecker) GetData() interface{} {
+	return l.data
+}
+
+// GetID returns the id of the section collected by the checker.
+func (l *HealthChecker) GetID() string {
+	return "health"
+}
+
+// GetTitle returns the title of the section collected by the checker.
+func (l *HealthChecker) GetTitle() string {
+	return "Installation health"
+}
+
+// isPodReady checks the conditions of the pod to determine if all the containers in the Pod are up and running.
+func (l *HealthChecker) isPodReady(conditions []corev1.PodCondition) bool {
+	// Check the condition of the pod
+	for _, c := range conditions {
+		if c.Type == corev1.PodReady {
+			return c.Status == corev1.ConditionTrue
+		}
+	}
+	return false
+}
```

Comment:
We can re-use this util function https://github.com/liqotech/liqo/blob/4992cda88235fd3c9da558bbfc348368c0463465/pkg/utils/pod/pod.go#L26